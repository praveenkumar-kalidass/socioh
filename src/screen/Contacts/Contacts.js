import React, { useEffect, useState } from 'react';
import { ScrollView } from 'react-native';
import PropTypes from 'prop-types';
import {
  Card,
  CardItem,
  Thumbnail,
  List,
  ListItem,
  Left,
  Text,
  Input,
  Item,
} from 'native-base';

import UserAvatar from '../../asset/image/user.png';
import { COLOR, ROUTE, TRANSLATION } from '../../constant';
import useService from '../../hook/useService';
import { ContactDetails, ContactsContainer } from './Contacts.style';

const Contacts = ({ navigation }) => {
  const [contacts, setContacts] = useState([]);
  const { getContacts } = useService();

  const loadContacts = async (search) => {
    try {
      const contactList = await getContacts(search);
      setContacts(contactList);
    } catch (error) {
      console.warn('Error:', error);
    }
  };

  useEffect(() => {
    loadContacts();
  }, []);

  const navigateToContact = (contact) => {
    navigation.navigate(ROUTE.CONTACT, { contact });
  };

  return (
    <ContactsContainer>
      <If condition={Boolean(contacts.length)}>
        <Card>
          <CardItem header>
            <Item regular rounded style={{ borderColor: COLOR.PRIMARY }}>
              <Input
                testID="contact_search_input"
                placeholder={TRANSLATION.SEARCH}
                onChangeText={loadContacts}
              />
            </Item>
          </CardItem>
          <ScrollView>
            <List>
              <For each="contact" index="index" of={contacts}>
                <ListItem
                  avatar
                  key={`contact_${index}`}
                  onPress={() => navigateToContact(contact)}>
                  <Left>
                    <Thumbnail small source={UserAvatar} />
                  </Left>
                  <ContactDetails>
                    <Text testID={`contact_name_${index}`}>
                      {`${contact.givenName} ${contact.familyName}`}
                    </Text>
                    <For
                      each="number"
                      index="numberIndex"
                      of={contact.phoneNumbers}>
                      <Text
                        note
                        testID={`phone_number_${index}_${numberIndex}`}
                        key={`phone_number_${index}_${numberIndex}`}>
                        {number.number}
                      </Text>
                    </For>
                  </ContactDetails>
                </ListItem>
              </For>
            </List>
          </ScrollView>
        </Card>
      </If>
    </ContactsContainer>
  );
};

Contacts.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
};

export default Contacts;
