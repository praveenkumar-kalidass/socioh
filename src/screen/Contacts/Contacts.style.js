import styled from 'styled-components/native';
import { Body } from 'native-base';

import { SPACING } from 'constant';

export const ContactsContainer = styled.View`
  flex: 1;
  padding-horizontal: ${SPACING.SPACE_20};
`;

export const ContactDetails = styled(Body)`
  border-bottom-width: 0;
`;
