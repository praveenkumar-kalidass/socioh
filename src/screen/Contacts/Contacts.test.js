import React from 'react';
import { render } from '@testing-library/react-native';

import Contacts from './index';

describe('Contacts', () => {
  it('should match snapshot', () => {
    const container = render(<Contacts />);

    expect(container).toMatchSnapshot();
  });
});
