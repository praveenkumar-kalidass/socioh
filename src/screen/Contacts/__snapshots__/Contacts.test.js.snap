// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts should match snapshot 1`] = `null`;
